source/collimation.f90

```diff
@@ -1331,46 +1176,446 @@ subroutine collimate_parseInputLine(inLine, iLine, iErr)
+
+  case("CERN")
+    if(nSplit /= 2) then
+      write(lerr,"(a,i0)") "COLL> ERROR CERN expects 1 value, got ",nSplit-1
+      write(lerr,"(a)")    "COLL>       CERN true|false"
+      iErr = .true.
+      return
+    end if
+    call chr_cast(lnSplit(2), cern, iErr)
+
+  case("SIGSECUT")
+    if(nSplit /= 3) then
+      write(lerr,"(a,i0)") "COLL> ERROR SIGSECUT expects 2 values, got ",nSplit-1
+      write(lerr,"(a)")    "COLL>       SIGSECUT sigma_xy sigma_r"
+      iErr = .true.
+      return
+    end if
+    call chr_cast(lnSplit(2), sigsecut2, iErr)
+    call chr_cast(lnSplit(3), sigsecut3, iErr)
+
+  case default
+    write(lerr,"(a)") "COLL> ERROR Unknown keyword '"//trim(lnSplit(1))//"'"
+    write(lerr,"(a)") "COLL>       The parser is assuming you want keyword/value block parsing as the first line had one value."
+    write(lerr,"(a)") "COLL>       The two formats cannot be mixed."
+    iErr = .true.
+    return
+
+  end select
+
+  return
+
+  !  Parse old style COLL block
+  ! ============================
+10 continue
+
   select case(iLine)
 
   case(1)
-
     if(nSplit /= 1) then
       write(lerr,"(a,i0)") "COLL> ERROR Expected 1 value on line 1, got ",nSplit
       iErr = .true.
       return
     end if
-
-    if(nSplit > 0) call chr_cast(lnSPlit(1),do_coll,iErr)
+    if(nSplit > 0) call chr_cast(lnSplit(1),do_coll,iErr)
 
   case(2)
-
-    if(nSplit > 0) call chr_cast(lnSPlit(1),nloop,iErr)
-    if(nSplit > 1) call chr_cast(lnSPlit(2),myenom,iErr)
-
-    if(nloop /= 1) then
-      write(lerr,"(a,i0)") "COLL> ERROR Support for multiple samples is deprecated. nloop must be 1, got ",nloop
+    if(nSplit /= 2) then
+      write(lerr,"(a,i0)") "COLL> ERROR Expected 2 values on line 2, got ",nSplit
       iErr = .true.
       return
     end if
+    if(nSplit > 0) call chr_cast(lnSplit(1),nloop,iErr)
+    if(nSplit > 1) call chr_cast(lnSplit(2),myenom,iErr)
 
-    if(napx*2 > npart) then
-      write(lerr,"(2(a,i0))") "COLL> ERROR Maximum number of particles is ", npart, ", got ",(napx*2)
+    if(nloop /= 1) then
+      write(lerr,"(a,i0)") "COLL> ERROR Multiple samples is no longer supported. nloop must be 1, got ",nloop
       iErr = .true.
       return
-   endif
+    end if
 
   case(3)
-    if(nSplit > 0)  call chr_cast(lnSPlit(1), do_thisdis,  iErr)
-    if(nSplit > 1)  call chr_cast(lnSPlit(2), mynex,       iErr)
-    if(nSplit > 2)  call chr_cast(lnSPlit(3), mdex,        iErr)
-    if(nSplit > 3)  call chr_cast(lnSPlit(4), myney,       iErr)
-    if(nSplit > 4)  call chr_cast(lnSPlit(5), mdey,        iErr)
-    if(nSplit > 5)  filename_dis = lnSPlit(6)
-    if(nSplit > 6)  call chr_cast(lnSPlit(7), enerror,     iErr)
-    if(nSplit > 7)  call chr_cast(lnSPlit(8), bunchlength, iErr)
+    if(nSplit /= 8) then
+      write(lerr,"(a,i0)") "COLL> ERROR Expected 8 values on line 3, got ",nSplit
+      iErr = .true.
+      return
+    end if
+    if(nSplit > 0)  call chr_cast(lnSplit(1), do_thisdis,  iErr)
```

Comment:
do we really need the `id (nSplit>` if we check that we get all the fields?

Reply:
No, not any more. Will clean up.